### DISCUSSION TITLE
Which IPs do I need to allow for a self-hosted runner?

### DISCUSSION THREAD

**Discussion by avik-so** [community/community#25107] [Actions]
<p>Where can I find the IPs and ports which I need to have open on our self hosted GitHub action runner to be able to properly communicate with GitHub?</p>
<p>Thanks</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Thanks <a class="mention" href="/avik-so">@avik-so</a> . The runners connect back to GitHub, make sure it’s connected to network and only outbound https 443 is needed.</p>
<p>You can get the exact ip in workflow with command ‘ipconfig’(windows) or ‘ifconfig’(linux/macOS).</p>
<p>Please refer to another <a href="https://github.community/t/which-ports-does-a-self-hosted-runner-need/17022">ticket</a> for more details.</p>

**Comment by avik-so** [comment]
<p>Maybe this is so simple that I’m missing something, but I just want to clarify.</p>
<p>When a GitHub workflow is activated, there is no call made to the runner to start the action?<br>
Instead, the runner regularly polls GitHub through an https call, and receives back data on what to run? Is that correct?</p>

**Comment by weide-zhou** [comment]
<p>Yes, you are correct! :slight_smile:</p>
